service/history/timerQueueStandbyTaskExecutor.go

```diff
@@ -518,6 +567,39 @@ func (t *timerQueueStandbyTaskExecutor) fetchHistoryFromRemote(
 	return consts.ErrTaskRetry
 }
 
+func (t *timerQueueStandbyTaskExecutor) pushActivity(
+	task tasks.Task,
+	postActionInfo interface{},
+	logger log.Logger,
+) error {
+
+	if postActionInfo == nil {
+		return nil
+	}
+
+	pushActivityInfo := postActionInfo.(*pushActivityTaskToMatchingInfo)
+	activityScheduleToStartTimeout := &pushActivityInfo.activityTaskScheduleToStartTimeout
+	activityTask := task.(*tasks.ActivityRetryTimerTask)
+	ctx, cancel := context.WithTimeout(context.Background(), transferActiveTaskDefaultTimeout)
+	defer cancel()
+
+	_, err := t.matchingClient.AddActivityTask(ctx, &matchingservice.AddActivityTaskRequest{
+		NamespaceId:       activityTask.NamespaceID,
+		SourceNamespaceId: activityTask.NamespaceID,
```

Comment:
I don't think they should be the same. `activityTask.NamespaceID` is the sourceNamespaceID.